OPHD/States/MapViewStateTurn.cpp

```diff
@@ -412,21 +407,22 @@ void MapViewState::updateBiowasteRecycling()
 }
 
 
-void MapViewState::countFood()
+void MapViewState::produceFood()
 {
-	mFood = 0;
+	for (auto& foodProducer : enumerateComponent<FoodProduction>())
+	{
+		foodProducer.produce();
+	}
+}
 
-	auto structures = NAS2D::Utility<StructureManager>::get().structureList(Structure::StructureClass::FoodProduction);
-	auto& command = NAS2D::Utility<StructureManager>::get().structureList(Structure::StructureClass::Command);
 
-	structures.insert(structures.begin(), command.begin(), command.end());
+void MapViewState::countFood()
+{
+	mFood = 0;
 
-	for (auto structure : structures)
+	for (auto& foodProducer : enumerateComponent<FoodProduction>())
 	{
-		if (structure->operational() || structure->isIdle())
-		{
-			mFood += static_cast<FoodProduction*>(structure)->foodLevel();
-		}
+		mFood += foodProducer.foodLevel();
```

Comment:
the operational/idle checks were superfluous, since the level was set to zero if the structure is disabled anyway.